### PR TITLE
fix: set system language as initial language

### PR DESCRIPTION
**Pull request by paul-boegelsack**
## Summary

The App still uses english as initial language. This RP changes the initial selection to the system language. 

## Changelog

```
- Selects the navigator language when the app is started for the first time
```

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [x] iOS
  - [x] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/mobile/App.svelte [hunk 1]
```diff
@@ -122,6 +120,9 @@
         await getVersionDetails()
         await pollMarketData()
         await pollNetworkStatus()
+
+        const initialLocale = $appRoute === AppRoute.Welcome ? null : $appSettings.language
```
Comment (amadeu2) [inline packages/mobile/App.svelte:124]: I think still we need a way to detect the device language, maybe we could use [device.getlanguagecode()](https://capacitorjs.com/docs/apis/device#getlanguagecode)
With the capacitor v4 we also could use `device#getlanguageTag` to process also the region code and possibly apply to the amount input to handle the `,.` issue

**Review by amadeu2 — COMMENTED** [review]
When tested with a clean install I still getting English as default, having Spanish defined on my device, maybe using the capacitor device plugin could help us to fix it

File: packages/mobile/App.svelte (hunk 1 above)
Comment (paul-boegelsack) [reply packages/mobile/App.svelte:124]: > When tested with a clean install I still getting English as default, having Spanish defined on my device, maybe using the capacitor device plugin could help us to fix it

Yeah, seems there is still an issue. It works on my iOS device but not on may Android it seems, just checked again. Do you have the problem on both OS?

Comment (amadeu2) [reply packages/mobile/App.svelte:124]: To reproduce on iOS we need to add a new language and keep English or any other there should be more than 1, then uninstall the app, change the language and install again. The JS way to get the language returns the first language of the list, but not the selected one on the OS. With Capacitor device plugin we get the correct selected language

**Review by amadeu2 — CHANGES_REQUESTED** [review]
I think adding only the [device.getlanguagecode()](https://capacitorjs.com/docs/apis/device#getlanguagecode) would be enough, forget by now the Capacitor v4 update, I've been testing on this branch some time ago but we'll need to check some issues https://github.com/iotaledger/firefly/tree/chore/update-capacitor-v4

File: packages/mobile/App.svelte (hunk 1 above)
Comment (paul-boegelsack) [reply packages/mobile/App.svelte]: > I think adding only the [device.getlanguagecode()](https://capacitorjs.com/docs/apis/device#getlanguagecode) would be enough, forget by now the Capacitor v4 update, I've been testing on this branch some time ago but we'll need to check some issues https://github.com/iotaledger/firefly/tree/chore/update-capacitor-v4

Alright, I added the getLanguageCode function and it seems to work know 👍 Could you check again? 😄

Comment (amadeu2) [reply packages/mobile/App.svelte]: now works! but seems there is still some issue with 'es' and 'pt' lang codes, I'll fix it

**Review by amadeu2 — APPROVED** [review]
Tested and working nice on Android and iOS 🥳 fixed specific cases on 'es' and 'pt' region codes

**Pull request merged**